### PR TITLE
Upgrade development dependencies and test matrix with latest Ruby and Rails versions

### PR DESCRIPTION
**Pull request by pkuczynski**

### CODE REVIEW

File: config.gemspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Please consider donating to our open collective to help us maintain this project
   s.required_ruby_version = '>= 2.4.0'
 
   s.add_dependency 'deep_merge', '~> 1.2', '>= 1.2.1'
-  s.add_dependency 'dry-validation', '~> 1.0'
+  s.add_dependency 'dry-validation', '~> 1.4.0'
```
Comment (jcoyne) [inline config.gemspec]: Why are you pinning so tightly here? A comment would be helpful.

File: config.gemspec [hunk 2]
```diff
@@ -47,10 +47,10 @@ Please consider donating to our open collective to help us maintain this project
   s.add_development_dependency 'mdl', '~> 0.8', '>= 0.8.0'
 
   # Version 0.62 requires Ruby 2.2
```
Comment (jcoyne) [inline config.gemspec]: Remove?

File: config.gemspec (hunk 1 above)
Comment (pkuczynski) [reply config.gemspec]: Do you think `s.add_dependency 'dry-validation', '~> 1.4'` would be better?

File: config.gemspec (hunk 2 above)
Comment (pkuczynski) [reply config.gemspec]: Indeed :)

File: config.gemspec (hunk 1 above)
Comment (jcoyne) [reply config.gemspec]: Yes, much better. Thanks

**Pull request merged**